challenges/backend/src/app/services/CarOnSaleClient/classes/Calculator.spec.ts

```diff
@@ -0,0 +1,96 @@
+import { expect } from 'chai';
+import "reflect-metadata";
+import { AuctionItem } from '../dtos/auction/AuctionItem';
+
+import { Calculator } from './Calculator';
+
+describe('Calculator', () => {
+  const calculator = new Calculator();
+
+  describe('calculatePercentage', () => {
+    it('should calculate the percentage correctly', () => {
+      const minimumRequiredAsk = 100;
+      const currentHighestBidValue = 50;
+      const expectedPercentage = 50;
+
+      const actualPercentage = calculator.calculatePercentage(currentHighestBidValue, minimumRequiredAsk);
+
+      expect(actualPercentage).to.equal(expectedPercentage);
+    });
+
+    it('should return 0 if minimumRequiredAsk is null', () => {
+      const minimumRequiredAsk = null;
+      const currentHighestBidValue = 50;
+      const expectedPercentage = 0;
+
+      const actualPercentage = calculator.calculatePercentage(currentHighestBidValue, minimumRequiredAsk);
+
+      expect(actualPercentage).to.equal(expectedPercentage);
+    });
```

Comment:
If you change minimumRequiredAsk to a numeric zero. You will receive an assertion error:
`AssertionError: expected Infinity to equal +0
Expected :0
Actual: Infinity`

You could add another test case to check that.